app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -131,6 +127,11 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
     }
 
     override fun onLoadingStateChanged(session: Session, loading: Boolean) {
+        if (!loading && session.url == APP_URL_POCKET_ERROR) {
+            browserOverlay?.showMegaTileError()
+            setOverlayVisible(true)
```

Comment:
This should live in the PocketViewModel, not the view layer.

Reply:
I'm about confused about this- how would this fit into the VM?

Reply:
@Baron-Severin TODO: overlay refactoring perhaps? not sure about exposing presentation logic in VM

Reply:
If you look at `BrowserNavigationOverlay#observeForMegaTile`, whenever it receives a `PocketViewModel.State.Error` event it displays an error megatile.  If we handle this logic in the view layer, our state will get out of sync.  (e.g., the VM might be in state `Feed`, but we've changed it by hand in the view layer to display an error).

Since we've moved Pocket over to MVVM, we want our VM to match our display state as closely as possible.  This means that, instead of putting this logic in the view layer, we should find a way to make the VM emit an `Error` object when one needs to be displayed.

Right now, that will mean adding a temporary method to the PocketVM that sends it the current URL.  Once the `SessionRepo` is in master, it can get it from there and we can remove the temporary method.

Reply:
@psymoon I'm not sure I follow?  Presentation logic is the VM's job.  It's supposed to mirror display state as closely as possible, and in this case that means that `PocketVM.state` should be `Error` whenever an error is displayed.

Reply:
@Baron-Severin yes, I agree 100% on how vs. what to display logic :)

Reply:
Actually, are we sure we need this code at all? As far as I'm aware, the only thing that can put Pocket into an error state is a failed network call, and the repo already handles that case.  Am I missing an edge case?

Reply:
This part is just for testing- QA wanted a link that would show the error screen for screenshot tests (#1111).

Reply:
We have a way to push mocked state to the `PocketVideoRepo` directly in Espresso tests now (`FirefoxTestApplication#pushPocketRepoState`).  If @npark-mozilla and @AaronMT are fine with it, I would like to remove the requirement to show a Pocket error on a specific URL, and use that method instead.

Reply:
Sure. PocketErrorTest.java would need an accompanying update